### DISCUSSION TITLE
Does this project support jsx, tsx?

### DISCUSSION THREAD

**Discussion by NieZhuZhu** [pmd/pmd#3092] [Q&A] [answered]
Does this project support jsx, tsx?

**Comment by jsotuyod** [comment, marked as answer]
No, currently there is no support for those. If a grammar is available it could be added [as a new PMD language module](https://pmd.github.io/pmd-6.31.0/pmd_devdocs_major_adding_new_language.html). Contributions would be most welcomed.
